### PR TITLE
Introduce new api route for resend invitation mail

### LINKED ISSUE #38954: [QA] resent invitation email invalidates original invitation link
## Seen in 10.8.0 beta2

* admin creates a new user account and only specifies username and email.
* the system sends out an email to the user 
image
* admin hits the mail envelope icon near the trashbin. Another email is generated with a different link.
* before the first invitaiton timeout expires, user attends to his mailbox and chooses the link from the first invitation email. It does not work. BAD.

Expected behaviour:  the re-sent mail has the same link. The expiration is extended accordingly.

### LINKED ISSUE #38955: [QA] message confirms invitation mail as sent, even when mailer is not configured
Seen in 10.8.0 bets2

Try this on a docker, where no mailer is configured:

* admin creates a new user (specifying an email address
* admin gets a warning, that the "password reset mail" (check that wording) could not be sent. OK
* admin tries to resend the invitation mail. He sees a friednly send confirmation:
image
* in reality no mail was sent. BAD.

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Introduce new api route instead of re-using `/resend/token/{userId}` since we need a correct http status response to indicate errors, as well we just want to make this route accessible by admins 

## Related Issue

- Fixes #38955 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by micbar — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by JanAckermann** [comment]
@phil-davis Code coverage .... what kind of test do you recommend? unit test?

**Comment by phil-davis** [comment]
> @phil-davis Code coverage .... what kind of test do you recommend? unit test?

SonarCloud is calculating unit test coverage. The new `resendInvitation` will need unit test coverage to keep SonarCloud happy - and it needs to be tested anyway.

**Comment by JanAckermann** [comment]
Ok tomorrow then :) :+1:

**Comment by JanAckermann** [comment]
Also added a condition, that the email will be only sent if the user never logged in.

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
+{"version":1,"defects":{"Tests\\Settings\\Controller\\UsersControllerTest::testResendInvitation":4,"Tests\\Settings\\Controller\\UsersControllerTest::tearDownAfterClass":3},"times":{"Tests\\Settings\\Controller\\UsersControllerTest::testResendInvitation":0.008,"Tests\\Settings\\Controller\\UsersControllerTest::tearDownAfterClass":0}}
```
Comment (phil-davis) [inline .phpunit.result.cache:1]: This is an accident.

`.gitignore` has `/tests/.phpunit.result.cache` - but maybe this cache comes in the root dir nowadays?
And we should add that to `.gitignore`

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt2-10.6.0-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31054/164/1

**Comment by JanAckermann** [comment]
@phil-davis made it general

File: .gitignore [hunk 2]
```diff
@@ -135,7 +134,7 @@ vendor-bin/**/composer.lock
 clover.xml
 /tests/output
 .php-cs-fixer.cache
-
+.phpunit.result.cache
```
Comment (phil-davis) [inline .gitignore:137]: That's good - it ignores `.phpunit.result.cache` anywhere in the repo.

Comment (JanAckermann) [reply .gitignore:137]: So next step is pray to the CI goods, I really want to close this issue :+1:

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-git-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31055/159/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-latest-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31055/160/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt2-git-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31055/162/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-10.6.0-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31055/161/1

**Comment by phil-davis** [comment]
More stupid drone-docker-pull errors! I restarted CI.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares1-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31056/115/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliExternalStorage-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31056/127/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares2-latest-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31056/117/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIProxySmoke-3-3-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31059/170/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares1-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31060/113/1

**Comment by phil-davis** [comment]
I restarted CI - maybe the silly docker pull problems will go away.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliExternalStorage-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31064/125/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares2-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31064/118/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares1-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31066/115/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot2-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31066/110/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliExternalStorage-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31068/127/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliExternalStorage-latest-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31068/126/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares2-latest-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31069/117/1

**Pull request merged**